include/net/lwm2m.h

```diff
@@ -200,6 +199,24 @@ typedef int (*lwm2m_engine_set_data_cb_t)(uint16_t obj_inst_id,
  */
 typedef int (*lwm2m_engine_user_cb_t)(uint16_t obj_inst_id);
 
+/**
+ * @brief Asynchronous execute notification callback.
+ *
+ * Resource executes trigger a callback of this type.
+ *
+ * Register a function of this type via:
+ * lwm2m_engine_register_exec_callback()
+ *
+ * @param[in] obj_inst_id Object instance ID generating the callback.
+ * @param[in] payload Pointer to data argument payload.
```

Comment:
I'd add information here, that it can be NULL if no arguments were provided in the execute message.